sharedbuffers/mapped_struct.py

```diff
@@ -385,26 +435,40 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
                 except:
                     isbuffer = False
                 if cython.compiled and isbuffer:
-                    cbuf[offs] = 'm'
-                    for i in xrange(1, 8):
-                        cbuf[offs+i] = 0
+                    if maxval < 56:
+                        cbuf[offs] = 'm'
+                        for i in xrange(1, 8):
+                            cbuf[offs+i] = 0
+                    else:
+                        cbuf[offs] = 'M'
+                        for i in xrange(1, 15):
+                            cbuf[offs+i] = 0
                     for ix in obj:
                         cbuf[offs+1+ix/8] |= 1 << (ix & 7)
                 else:
-                    buf[offs:offs+1] = 'm'
-                    buf[offs+1:offs+8] = '\x00\x00\x00\x00\x00\x00\x00'
+                    if maxval < 56:
+                        buf[offs:offs+1] = 'm'
+                        buf[offs+1:offs+8] = '\x00\x00\x00\x00\x00\x00\x00'
+                    else:
+                        buf[offs:offs+1] = 'M'
+                        buf[offs+1:offs+15] = '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```

Comment:
I think there's a mismatch here.
When ```maxval < 56```, you write 7 bytes, using offsets 1 to 8
When ```maxval >= 56```, you write 15 bytes, using offsets 1 to 15

I think in the second case, you have to use offsets 1 to 16 (And also update offs accordingly below).